docs/how-to-guides/dataset-selection.py

```diff
@@ -0,0 +1,216 @@
+# ---
+# jupyter:
+#   jupytext:
+#     text_representation:
+#       extension: .py
+#       format_name: percent
+#       format_version: '1.3'
+#       jupytext_version: 1.16.4
+#   kernelspec:
+#     display_name: Python 3 (ipykernel)
+#     language: python
+#     name: python3
+# ---
```

Comment:
I couldn't run this notebook because my `data_catalog` is empty. Is there some set-up that I should do before I can run it?

Reply:
You should pull the data needed for the tests. This will setup esgpull in the current directory

```
uv run esgpull self install $PWD/.esgpull
make fetch-test-data
```

There isn't a super happy path to getting the local config setup for the REF yet, but what I tend to do is:

```
uv run ref config list > $PWD/.ref/ref.toml
export REF_CONFIGURATION=$PWD/.ref
```
then edit `$PWD/.ref/ref.toml` to fix the paths and database url (they will default to somewhere in your home directory) and set `paths.allow_out_of_tree_datasets` to true. This does three things, creates an in-source config file,  sets the `ref` cli to use the new config and allows for datasets to be ingested from outside of `paths.data`.

My ref.toml file for reference:
```
[paths]
data = "/Users/jared/code/CMIP-REF/cmip-ref/.ref/data"
log = "/Users/jared/code/CMIP-REF/cmip-ref/.ref/log"
tmp = "/Users/jared/code/CMIP-REF/cmip-ref/.ref/tmp"
allow_out_of_tree_datasets = true

[db]
database_url = "sqlite:////Users/jared/code/CMIP-REF/cmip-ref/.ref/db/ref.db"
run_migrations = true
```

Then ingest the downloaded datasets into the database
```
uv run ref ingest --source-type cmip6 $PWD/.esgpull/data
```

On the first run this will create a sqlite file according to `db.database_url` and apply any migrations, before walking the directory provided to find that available datasets which are then added to the db.

Reply:
Nice, thanks, that worked. Is it worth including this somewhere in the developer docs?

Reply:
Yup definitely. I've made `allow_out_of_tree_datasets` default to true for now to make it easier setup to the test suite. This should probably be false for users but we are a while away from worrying about the happy path for users